torchvision/prototype/transforms/kernels/_geometry.py

```diff
@@ -0,0 +1,76 @@
+from typing import Tuple, List, Optional, TypeVar
+
+import torch
+from torchvision.prototype import features
+from torchvision.transforms import functional as _F, InterpolationMode
+
+from ._meta_conversion import convert_bounding_box_format
+
+
+T = TypeVar("T", bound=features.Feature)
+
+
+horizontal_flip_image = _F.hflip
+
+
+def horizontal_flip_bounding_box(
+    bounding_box: torch.Tensor, *, format: features.BoundingBoxFormat, image_size: Tuple[int, int]
+) -> torch.Tensor:
+    shape = bounding_box.shape
+
+    bounding_box = convert_bounding_box_format(
+        bounding_box, old_format=format, new_format=features.BoundingBoxFormat.XYXY
+    ).view(-1, 4)
+
+    bounding_box[:, [0, 2]] = image_size[1] - bounding_box[:, [0, 2]]
+
+    return convert_bounding_box_format(
+        bounding_box, old_format=features.BoundingBoxFormat.XYXY, new_format=format
+    ).view(shape)
+
+
+def resize_image(
+    image: torch.Tensor,
+    size: List[int],
+    interpolation: InterpolationMode = InterpolationMode.BILINEAR,
+    max_size: Optional[int] = None,
+    antialias: Optional[bool] = None,
+) -> torch.Tensor:
+    new_height, new_width = size
+    num_channels, old_height, old_width = image.shape[-3:]
+    batch_shape = image.shape[:-3]
+    return _F.resize(
+        image.reshape((-1, num_channels, old_height, old_width)),
+        size=size,
+        interpolation=interpolation,
+        max_size=max_size,
+        antialias=antialias,
+    ).reshape(batch_shape + (num_channels, new_height, new_width))
+
+
+def resize_segmentation_mask(
+    segmentation_mask: torch.Tensor,
+    size: List[int],
+    interpolation: InterpolationMode = InterpolationMode.NEAREST,
+    max_size: Optional[int] = None,
+    antialias: Optional[bool] = None,
+) -> torch.Tensor:
+    return resize_image(
+        segmentation_mask, size=size, interpolation=interpolation, max_size=max_size, antialias=antialias
```

Comment:
shouldn't we disable the `interpolation` altogether here?

Otherwise if the user passes `interpolation=BICUBIC` in the main `Resize()` transform, this will be using bicubic interpolation unwillingly?

Reply:
@fmassa Ah good catch thanks for flagging. 
@pmeier The mask is basically a mapping of label values. We can't linearly interpolate between ids. Only Nearest makes sense here.